src/MerchantCenter/PolicyComplianceCheck.php

```diff
@@ -68,6 +70,116 @@ public function is_accessible(): bool {
 		return true;
 	}
 
+	/**
+	 * Check if the store sample product landing pages lead to a 404 error.
+	 *
+	 * @return bool
+	 */
+	public function has_page_not_found_error(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headers ) && $headers[0] !== 'HTTP/1.1 200 OK' ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check if the store sample product landing pages has redirects through 3P domains.
+	 *
+	 * @return bool
+	 */
+	public function has_redirects(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headesr ) && isset( $headers[0] ) ) {
+			if ( $headers[0] === 'HTTP/1.1 302 Found' ) {
+				// this is the URL where it's redirecting
+				return true;
+			}
+		}
+			return false;
+	}
+
+	/**
+	 * Get the headers of the store sample product landing pages.
+	 *
+	 * @return array
+	 */
+	public function get_landing_page_headers(): array {
+		$products = wc_get_products(
+			[
+				'limit' => 1,
+			]
+		);
+		if ( ! empty( $products ) ) {
+			$response = wp_remote_get( $products[0]->get_permalink() );
+			if ( is_array( $response ) && ! is_wp_error( $response ) ) {
+				return $response['headers'];
+			}
+		}
+		return [];
+	}
+
+	/**
+	 * Check if the merchant set the restrictions in robots.txt or not in the store.
+	 *
+	 * @return bool
+	 */
+	public function has_restriction(): bool {
+		return ! $this->robots_allowed( $this->get_site_url() );
+	}
+
+	/**
+	 * Check if the robots.txt has restrictions or not in the store.
+	 *
+	 * @param string $url
+	 * @return bool
+	 */
+	private function robots_allowed( $url ) {
+		$agents = [ preg_quote( '*', '/' ) ];
+		$agents = implode( '|', $agents );
+
+		// location of robots.txt file
+		$robotstxt = file( trailingslashit( $url ) . 'robots.txt' );
```

Comment:
On my test server this request is blocked (unless I override the configuration), and there could be other server configurations which do not allow file access to an external URL. It's probably better to switch to a `wp_remote_get` since that's generally confirmed to work. Note that we'll need to break up the content into lines, if we want to loop through it line by line (which is done by default by the `file` function).